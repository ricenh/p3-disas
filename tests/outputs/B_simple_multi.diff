--- outputs/B_simple_multi.txt	2023-11-01 19:33:22.610533326 -0400
+++ expected/B_simple_multi.txt	2023-10-10 11:54:35.000000000 -0400
@@ -1,11 +1,17 @@
-Usage: ../y86 <option(s)> mini-elf-file
- Options are:
-  -h      Display usage
-  -H      Show the Mini-ELF header
-  -a      Show all with brief memory
-  -f      Show all with full memory
-  -s      Show the program headers
-  -m      Show the memory contents (brief)
-  -M      Show the memory contents (full)
-  -d      Disassemble code contents
-  -D      Disassemble data contents
+Disassembly of executable contents:
+  0x100:                               | .pos 0x100 code
+  0x100: 10                            |   nop
+  0x101: 10                            |   nop
+  0x102: 10                            |   nop
+  0x103:                               | _start:
+  0x103: 30 f0 68 24 00 00 00 00 00 00 |   irmovq 0x2468, %rax
+  0x10d: 30 f3 34 12 00 00 00 00 00 00 |   irmovq 0x1234, %rbx
+  0x117: 60 03                         |   addq %rax, %rbx
+  0x119: 70 00 02 00 00 00 00 00 00    |   jmp 0x200
+
+  0x200:                               | .pos 0x200 code
+  0x200: 30 f1 78 56 00 00 00 00 00 00 |   irmovq 0x5678, %rcx
+  0x20a: 30 f2 14 15 00 00 00 00 00 00 |   irmovq 0x1514, %rdx
+  0x214: 61 21                         |   subq %rdx, %rcx
+  0x216: 00                            |   halt
+
