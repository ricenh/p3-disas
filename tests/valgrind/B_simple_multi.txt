==1079518== Memcheck, a memory error detector
==1079518== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==1079518== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==1079518== Command: ../y86 -d inputs/simple_multi.o
==1079518== 
Disassembly of executable contents:
  0x100:                               | .pos 0x100 code
  0x100: 10                            |   nop
  0x101: 10                            |   nop
  0x102: 10                            |   nop
  0x103:                               | _start:
  0x103: 30 f0 68 24 00 00 00 00 00 00 |   irmovq 0x2468, %rax
  0x10d: 30 f3 34 12 00 00 00 00 00 00 |   irmovq 0x1234, %rbx
  0x117: 60 03                         |   addq %rax, %rbx
  0x119: 70 00 02 00 00 00 00 00 00    |   jmp 0x200

  0x200:                               | .pos 0x200 code
  0x200: 30 f1 78 56 00 00 00 00 00 00 |   irmovq 0x5678, %rcx
  0x20a: 30 f2 14 15 00 00 00 00 00 00 |   irmovq 0x1514, %rdx
  0x214: 61 21                         |   subq %rdx, %rcx
  0x216: 00                            |   halt

==1079518== 
==1079518== HEAP SUMMARY:
==1079518==     in use at exit: 0 bytes in 0 blocks
==1079518==   total heap usage: 4 allocs, 4 frees, 20,952 bytes allocated
==1079518== 
==1079518== All heap blocks were freed -- no leaks are possible
==1079518== 
==1079518== For lists of detected and suppressed errors, rerun with: -s
==1079518== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
