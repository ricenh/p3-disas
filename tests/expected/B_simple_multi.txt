Disassembly of executable contents:
  0x100:                               | .pos 0x100 code
  0x100: 10                            |   nop
  0x101: 10                            |   nop
  0x102: 10                            |   nop
  0x103:                               | _start:
  0x103: 30 f0 68 24 00 00 00 00 00 00 |   irmovq 0x2468, %rax
  0x10d: 30 f3 34 12 00 00 00 00 00 00 |   irmovq 0x1234, %rbx
  0x117: 60 03                         |   addq %rax, %rbx
  0x119: 70 00 02 00 00 00 00 00 00    |   jmp 0x200

  0x200:                               | .pos 0x200 code
  0x200: 30 f1 78 56 00 00 00 00 00 00 |   irmovq 0x5678, %rcx
  0x20a: 30 f2 14 15 00 00 00 00 00 00 |   irmovq 0x1514, %rdx
  0x214: 61 21                         |   subq %rdx, %rcx
  0x216: 00                            |   halt

